packages/trace-viewer/src/ui/wsPort.ts

```diff
@@ -15,14 +15,36 @@
  */
 
 let lastId = 0;
-let _ws: WebSocket;
+let _ws: WebSocket | undefined;
 const callbacks = new Map<number, { resolve: (arg: any) => void, reject: (arg: Error) => void }>();
+let retryCount = 0;
+let pingInterval: NodeJS.Timeout | undefined;
 
-export async function connect(options: { onEvent: (method: string, params?: any) => void, onClose: () => void }): Promise<(method: string, params?: any) => Promise<any>> {
+export async function connect({
+  onEvent,
+  onClose = () => {},
+  maxRetries = 3,
+}: {
+  onEvent: (method: string, params?: any) => void;
+  onClose?: () => void;
+  maxRetries?: number;
+}): Promise<(method: string, params?: any) => Promise<any>> {
   const guid = new URLSearchParams(window.location.search).get('ws');
   const ws = new WebSocket(`${window.location.protocol === 'https:' ? 'wss' : 'ws'}://${window.location.hostname}:${window.location.port}/${guid}`);
+  window.ws = ws;
+  ws.addEventListener('close', () => {
```

Comment:
I see we lack `on('error')` handler as well. Should we copy logic from the WebSocketTransport?

Reply:
`close` always gets called, also in case of an error. Thats why I left it out, seems fine?